### ISSUE TITLE
Build: Redesign megamenu JS 

### ISSUE THREAD

**Issue by VictoriaSunNIH** [NCIOCPL/cgov-digital-platform#3277]
### Build the JS component of the redesign megamenu based on design specs/requirements documented in this ticket for pulling data into the menu

> **ESTIMATE** 8

This story is to create a CgdpMegaMenuAdapter for use in initializing the NCIExtendedHeaderWithMegaMenu component. This will use the API that is being created in #3265.  Below is a rough sequence diagram of how this will work.

NOTE: If a primary navigation item does NOT have megamenu content, then clicking on that item should go directly to that section and not open up an empty mega menu. This applies to Spanish Grants and Training specifically.

```mermaid
sequenceDiagram
    actor Editor
    actor Visitor
    Editor->>/taxonomy/term/32/edit: Enters YAML Content for Mega Menu in Drupal
    Drupal->>/taxonomy/term/32/edit: Validates YAML
    Drupal->>/taxonomy/term/32/edit: Stores YAML

    Visitor->>Web Page: load
    Web Page->>+NCIExtendedHeaderWithMegaMenu: create()
    Visitor->>Web Page: click About Cancer (id: 32)
    Web Page->>NCIExtendedHeaderWithMegaMenu: handle click event
    NCIExtendedHeaderWithMegaMenu->>cgdpMegaMenuAdapter: getMegaMenuContent(id: 32)
    cgdpMegaMenuAdapter-)/taxonomy/term/32/mega-menu: Fetch Data
    /taxonomy/term/32/mega-menu-->>cgdpMegaMenuAdapter: json data
    cgdpMegaMenuAdapter-->>cgdpMegaMenuAdapter: Parse json and create HTML
    cgdpMegaMenuAdapter->>NCIExtendedHeaderWithMegaMenu: HtmlElement
    NCIExtendedHeaderWithMegaMenu-->>Web Page: Open Mega Menu Overlay
    NCIExtendedHeaderWithMegaMenu-->>Web Page: Display Html from adapter
  
```
The ID passed into getMegamenuContent will come from the data-menu-id attribute on the navigation item.
Each navigation item in the primary nav should have a `data-menu-id` attribute with the ID of that Drupal site section.

### InVision: 
  * DSM link: *insert link here, or enter N/A if not needed*
  * Desktop mockup: *insert link here, or enter N/A if not needed*
  * Tablet mockup: *insert link here, or N/A if not needed*
  * Mobile mockup: *insert link here, or N/A if not needed*

### Acceptance Criteria for Behaviors 
*If a component changes between breakpoints, or things show/hide, that gets captured here* 

```gherkin
Scenario: User is activating the megamenu
  Given user is on c.gov on desktop
  When the user clicks on an L1 item in the main nav
  Then the megamenu for that L1 will appear below 
     And the L1 will be highlighted in the main nav
     And the caret for the L1 will flip to point upwards
  When the user hovers on any link in the megamenu
  Then the link will receive an underline
  When the user clicks on the L1 item that appears in the megamenu ("Explore L1")
  Then the landing page will load
     And the megamenu will close
     And an indicator will appear in the main nav under that L1
  When an L2 is clicked on 
  Then the page will load
     And the megamenu will close
     And an indicator will appear in the main nav under the L1 parent
  When an L3 is clicked on 
  Then the page will load
     And the megamenu will close
     And an indicator will appear in the main nav under the L1 parent
  When the user clicks on an L1 item in the main nav
  Then the megamenu for that L1 will appear below 
  When the user clicks outside of the megamenu overlay or on the L1 again
  Then the megamenu will close
  When the user clicks on an L1 item in the main nav
  Then the megamenu for that L1 will appear below 
  When another L1 is clicked on
  Then the megamenu for that L1 will appear below 
```

### Acceptance Criteria for Analytics
``` gherkin
Scenario:
Given a user is on any page of cancer.gov
When the user clicks on a primary nav link that opens the mega menu
Then there should be an NCIDataLayer Other event 
  And the event has the name “PrimaryNav:Open”
  And the event has the the linkname “PrimaryNav:Open”
  And this event should have the following data:
      | key          | value                                                          |
      | topLevelMenu | Top nav section menu of menu that is opened (ie: About Cancer) |
      | location     | Primary Nav                                                       |
```
``` gherkin
Scenario:
Given a user is on any page of cancer.gov
When the user clicks on a primary nav link that goes directly to the page (no mega menu)
Then there should be an NCIDataLayer Other event 
  And the event has the name “PrimaryNav:LinkClick”
  And the event has the linkname “PrimaryNav:LinkClick”
And have the following data:
    | key          | value                                                          |
      | location     | Primary Nav                                                     |     
      | primaryNavItem | Top nav section menu of menu that is opened (ie: About Cancer) |
      | listHeading  | This should pass the same value as the link text (the name of the button)               |
      | linkText     | Text of the link clicked                                       |
      | linkType | This would be "Primary Nav Button" for this type of click |
      | listHeadingNumber  | Should pass as 0 in this scenario.                  |
      | listItemNumber   | Should pass as 0 in this scenario.     |
```

``` gherkin
Given a user is on any page of cancer.gov
  And the mega menu has been opened
When the user clicks on a link in the mega menu
Then there should be an NCIDataLayer Other event 
  And the event has the name “PrimaryNav:LinkClick”
  And the event has the linkname “PrimaryNav:LinkClick”
And have the following data:
      | key          | value                                                          |
      | location     | Primary Nav                                                     |     
      | primaryNavItem | Top nav section menu of menu that is opened (ie: About Cancer) |
      | listHeading  | _Subheading of link clicked (ie: Advanced Cancer) . If link clicked is primary nav label (i.e. Explore About Cancer), this value should pass the linkText (i.e. Explore About Cancer)          |
      | linkText     | Text of the link clicked                                       |
      | linkType | This would be "List Heading", "List Item" or "Primary Nav Label" |
      | listHeadingNumber  | This is the number of the list heading within the larger list (ex: for About Cancer Menu, Causes & Prevention would be 2). If the user clicks on a primary nav label (ex: Explore About Cancer) this value should be set to 0.                  |
      | listItemNumber   | This is the number within a list heading of the links (ex: for About Cancer Menu, Cancer Statistics would be 2. If link clicked is a primary nav label or list heading this should be 0.     |
      
```
---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*

- [x] #3268
- [x] #3265
- [ ] https://github.com/NCIOCPL/ncids/issues/297

### Tasks
- [ ] Add `data-menu-id` to each of the items in the primary nav in the template

**Comment by bennettcc** [comment]
This has been approved through testing - but cannot be merged until the recent header changes (https://github.com/NCIOCPL/ncids/pull/545) are approved by the client and merged

**Comment by bennettcc** [comment]
6 SPs completed during S36, 2 remaining for S37

**Comment by sarinapadilla** [comment]
Closing this ticket as it has been fixed by PR #3449!
